### ISSUE TITLE
Add link the HalGet/SetBusDataByOffset functions to help users find the right method send the IRPs in the "Accessing PCI Device Configuration Space"

### ISSUE THREAD

**Issue by guliteb** [MicrosoftDocs/windows-driver-docs#3733]
Microsoft confirmed HalGet/SetBusDataByOffset functions are helper functions which send the IRPs detailed in the "Accessing PCI Device Configuration Space" document and we can also request to update this document to link the Hal functions to help users find them.

Please help to add link the Hal functions to help users find them.
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/nf-ntddk-halgetbusdatabyoffset
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/nf-ntddk-halsetbusdatabyoffset

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: bd0a490c-5642-3863-596f-a0e5d7ba9644
* Version Independent ID: f06461a3-83eb-7d2e-e3e7-dfcfd2e69fa8
* Content: [Accessing PCI Device Configuration Space - Windows drivers](https://learn.microsoft.com/en-us/windows-hardware/drivers/pci/accessing-pci-device-configuration-space)
* Content Source: [windows-driver-docs-pr/pci/accessing-pci-device-configuration-space.md](https://github.com/MicrosoftDocs/windows-driver-docs/blob/staging/windows-driver-docs-pr/pci/accessing-pci-device-configuration-space.md)
* Product: **windows-hardware**
* Technology: **pci**
* GitHub Login: @tedhudek
* Microsoft Alias: **tedhudek**

**Comment by tedhudek** [comment]
Thanks @guliteb for the feedback!  I've added a warning to https://learn.microsoft.com/en-us/windows-hardware/drivers/pci/accessing-pci-device-configuration-space (at the bottom).  I also cleaned up https://learn.microsoft.com/en-us/windows-hardware/drivers/kernel/accessing-device-configuration-space and added the links to the sample code.  Finally, added a warning to the top of https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/nf-ntddk-halgetbusdatabyoffset and https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/nf-ntddk-halsetbusdatabyoffset. 

The bottom line is we highly recommend using BUS_INTERFACE_STANDARD or IRP_MN_READ_CONFIG / IRP_MN_WRITE_CONFIG, but those two HAL routines are there for backward compatibility if those methods cannot be used.

Hope this helps!
